<a href="https://colab.research.google.com/github/agmorcillo/Coursera_Capstone/blob/main/New_York_city_population_and_demographic_data.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## **Problem Statement**







The objective is to locate and recommend to the client which neighbourhood in New York City will be best choice to start a Comic Shop.

New York city population and demographic data

Data source : https://en.wikipedia.org/wiki/New_York_City ; https://en.wikipedia.org/wiki/Brooklyn. Web scraping techniques was used to get NYC population density and demographics data from Wikipedia.


Web scraping techniques was used to get NYC population density and demographics data from Wikipedia.

In [2]:
import pandas as pd
import requests
from bs4 import BeautifulSoup

print('Libraries imported.')

Libraries imported.


In [3]:
response_obj = requests.get('https://en.wikipedia.org/wiki/New_York_City').text
soup = BeautifulSoup(response_obj,'lxml')
Neighborhoods_NYC_Table = soup.find('table', {'class':'wikitable sortable'})

# Preparation of the table - https://en.wikipedia.org/wiki/New_York_City 

In [4]:
rows = Neighborhoods_NYC_Table.select("tbody > tr")[3:8]

boroughs = []
for row in rows:
    borough = {}
    tds = row.select('td')
    borough["borough"] = tds[0].text.strip()
    borough["county"] = tds[1].text.strip()
    borough["population"] = float(tds[2].text.strip().replace(",",""))
    borough["gdp_billions"] = float(tds[3].text.strip().replace(",",""))
    borough["gdp_per_capita"] = float(tds[4].text.strip().replace(",",""))
    borough["land_sqm"] = float(tds[5].text.strip().replace(",",""))
    borough["land_sqkm"] = float(tds[6].text.strip().replace(",",""))
    borough["persons_sqm"] = float(tds[7].text.strip().replace(",",""))
    borough["persons_sqkm"] = float(tds[8].text.strip().replace(",",""))
    
    boroughs.append(borough)

print(boroughs)

[{'borough': 'The Bronx', 'county': 'Bronx', 'population': 1418207.0, 'gdp_billions': 42.695, 'gdp_per_capita': 30100.0, 'land_sqm': 42.1, 'land_sqkm': 109.04, 'persons_sqm': 33867.0, 'persons_sqkm': 13006.0}, {'borough': 'Brooklyn', 'county': 'Kings', 'population': 2559903.0, 'gdp_billions': 91.559, 'gdp_per_capita': 35800.0, 'land_sqm': 70.82, 'land_sqkm': 183.42, 'persons_sqm': 36147.0, 'persons_sqkm': 13957.0}, {'borough': 'Manhattan', 'county': 'New York', 'population': 1628706.0, 'gdp_billions': 600.244, 'gdp_per_capita': 368500.0, 'land_sqm': 22.83, 'land_sqkm': 59.13, 'persons_sqm': 71341.0, 'persons_sqkm': 27544.0}, {'borough': 'Queens', 'county': 'Queens', 'population': 2253858.0, 'gdp_billions': 93.31, 'gdp_per_capita': 41400.0, 'land_sqm': 108.53, 'land_sqkm': 281.09, 'persons_sqm': 20767.0, 'persons_sqkm': 8018.0}, {'borough': 'Staten Island', 'county': 'Richmond', 'population': 476143.0, 'gdp_billions': 14.514, 'gdp_per_capita': 30500.0, 'land_sqm': 58.37, 'land_sqkm': 15

**View of the table**

In [5]:
df = pd.DataFrame(boroughs, columns=["borough","county", "population", "gdp_per_capita", "persons_sqkm"])
df.head()

borough    county  population  gdp_per_capita  persons_sqkm
0      The Bronx     Bronx   1418207.0         30100.0       13006.0
1       Brooklyn     Kings   2559903.0         35800.0       13957.0
2      Manhattan  New York   1628706.0        368500.0       27544.0
3         Queens    Queens   2253858.0         41400.0        8018.0
4  Staten Island  Richmond    476143.0         30500.0        3150.0

# *Preparation of the table - https://en.wikipedia.org/wiki/Brooklyn*

In [6]:
table_bro = pd.read_html('https://en.wikipedia.org/wiki/Brooklyn')

In [7]:
print(f'Total tables: {len(table_bro)}')

Total tables: 26


## **Here we select the table that we want to visualize**

In [8]:
table_bro = pd.read_html('https://en.wikipedia.org/wiki/Brooklyn', match='Racial composition')
len(table_bro)

1

### **And we visualize it**

In [9]:
df = table_bro[0]
df.head()

Racial composition 2018[65]  ... 1950[67] 1900[67]
0                             White    49.5%  ...    92.2%    98.3%
1                     —Non-Hispanic    30.4%  ...      NaN      NaN
2         Black or African American    40.1%  ...     7.6%     1.6%
3  Hispanic or Latino (of any race)    19.1%  ...      NaN      NaN
4                             Asian    12.7%  ...     0.1%     0.1%

[5 rows x 6 columns]

### **Preliminary finding indicates that**

 - Brooklyn is the  most populous urban area in New York City (NYC)
 - However, with that being said, it is the most ethnically diverse urban area in NYC with the highest White ethnic population.
 - Brookly has a population density of 2559903 people per square km.